### ISSUE TITLE
Multiple Select Check Boxes with Grouped Options

### ISSUE THREAD

**Issue by testaj** [cakephp/cakephp#8385]
This is a (multiple allowed):
* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.1.10
* Platform and Target: PHP 5.4.28/MySQL 5.5.48.

### What you did
First, I've successfully implemented a Multiple Select box with option groups based on providing an providing a group of options structured like this:

```
    $options = [
       'Group 1' => [
          'Value 1' => 'Label 1',
          'Value 2' => 'Label 2'
       ],
       'Group 2' => [
          'Value 3' => 'Label 3'
       ]
    ];
    echo $this->Form->select('field', $options);
```

This is straight from the Cookbook at: http://book.cakephp.org/3.0/en/views/helpers/form.html#creating-select-pickers

It works great, and makes a select box exactly as it should.  My issue is that I would like to change the display to be multiple checkboxes.  In order to do that, I've switched the code to:

```
    echo $this->Form->input('field', [
            'multiple' => 'checkbox',
            'options' => $options
        ]);
```

### Expected Behavior
As indicated in the book, I would expect this code to create multiple checkboxes grouped by the group level in the $options array.  

### Actual Behavior
When I do this, the display ends up being a single checkbox, with all the options listed out next to it. 

I've found a thread on SO about this topic, and I've posted my own question.  Responses seem to indicate that this functionality is not currently built into the core.  I'm submitting this ticket as the functionality is written in the cookbook as part of the core, but does not work as outlined in the cookbook.

Thanks for your response to this issue.

**Comment by markstory** [comment]
I don't see where in the docs it indicates that `checkbox=multiple` and grouped options would work. You're right that this feature has not been implemented.

**Comment by testaj** [comment]
Hi Mark, it's included in the docs at this link:

http://book.cakephp.org/3.0/en/views/helpers/form.html#creating-select-pickers

It's under the title Creating Select Pickers.  Here's what it says:  

> If you would like to generate a select with optgroups, just pass data in hierarchical format. This works on multiple checkboxes and radio buttons too, but instead of optgroups wraps elements in fieldsets:

It's kind of pointless to argue bug vs enhancement.  It's a feature that would help some, and I noticed wasn't functioning as outlined in the docs.  

Thanks for helping!

**Comment by markstory** [comment]
Interesting. I don't remember fieldset generation for multiple-checkboxes. I'll update the docs so they aren't misleading at the very least.

**Comment by testaj** [comment]
That's perfect, just hoping to reduce confusion for others and maybe get something in the pipeline.  Thanks again.

**Comment by markstory** [comment]
I did some digging into the 2.x code and the documented behavior _did_ work in 2.x. However, there isn't a single test documenting the behavior. When I re-wrote the FormHelper for 3.x I relied on the test cases which is why this use case was missed. I'll look into getting it fixed for 3.3

**Comment by markstory** [comment]
Pull request up now.
